enhancements/agent-installer/interactive-installation-using-the-agent-based-installer-for-disconnected-environments.md

```diff
@@ -0,0 +1,237 @@
+itself.
+
+### Workflow Description
+
+This workflow is activated when the sentinel file, /etc/assisted/interactive-ui,
+is present in the Agent Based Installer ignition.
+
+**cluster creator** is a human user responsible for deploying a cluster.
+
+**Assisted Installer UI** is UI used to configure, initiate, and monitor the
+cluster installation. It is hosted on the rendezvous host.
+
+**rendezvous host** is a host on the cluster that serves the Assisted Service and 
+Assisted installer UI. It also serves as the bootstrap node before it reboots
+during cluster installation to become a member of the cluster.
+
+**OCP console** is the installed cluster's UI.
+
+1. The cluster creator obtains or generates an ISO and boots it on target
+cluster hosts.
+
+2. The URL to the Assisted Installer UI running on a predefined port is
+displayed on the console.
+   
+3. On a separate machine the cluster creator opens up the browser to the
+Assisted Installer UI running on the rendezvous host.
+   
+4. The cluster creator begins configuring the cluster through the Assisted
+Installer UI, by providing all the required details.
+	
+5. If there are no configuration issues, the cluster creator moves to the 
+download credentials screen and downloads files containing the kubeconfig and
+kubeadmin password.
+	
+6. Then the Assisted Installer UI shows a confirmation screen showing the 
+configuration that was entered and a list of operators that will be enabled.
+The cluster creator clicks on a button to begin the installation.
+	
+7. The Assisted Installer UI shows the installation progress. When the
+installation reaches the point when the rendezvous host is rebooted and will
+join the cluster as a node, the UI will display the URL to the OCP console.
+	
+8. When ready, users may access the OCP console.
+
+### API Extensions
+
+There are no changes to APIs.
+
+### Topology Considerations
+
+Will function identically in all topologies.
+
+#### Hypershift / Hosted Control Planes
+
+There are no specific considerations for this feature regarding
+Hypershift / Hosted Control Planes.
+
+#### Standalone Clusters
+
+There are no specific considerations for this feature regarding 
+Standalone Clusters.
+
+#### Single-node Deployments or MicroShift
+
+The same workflow applies to SNO. 
+
+There is no specific considerations for this feature regarding
+MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+No additional details.
+
+### Risks and Mitigations
+
+Communications between the agent running on all nodes and assisted-service
+running on the redenzvous host are currently unsecured. Authentication and
+authorization will need to be added in the future.
+
+### Drawbacks
+
+None.
+
+## Alternatives (Not Implemented)
+
+None.
+
+## Test Plan
+
+The entire installation flow from creation of the boot ISO to initiating the
+install through the Assisted Installer UI and verifying the cluster and
```

Comment:
Just wondering, how the UI flow will be tests in e2e? I.e. some automation for the interactive part?
Or the configuration values will simply be passed using some files in tests?